tests/integration/butterfree/pipelines/test_feature_set_pipeline.py

```diff
@@ -129,3 +130,179 @@ def test_feature_set_pipeline(
 
         # tear down
         shutil.rmtree("test_folder")
+
+    def test_feature_set_pipeline_with_dates(
+        self,
+        mocked_date_df,
+        spark_session,
+        fixed_windows_output_feature_set_date_dataframe,
+    ):
+        # arrange
+        table_reader_id = "b_source"
+        table_reader_table = "b_table"
+        table_reader_db = environment.get_variable("FEATURE_STORE_HISTORICAL_DATABASE")
+        create_temp_view(dataframe=mocked_date_df, name=table_reader_id)
+        create_db_and_table(
+            spark=spark_session,
+            table_reader_id=table_reader_id,
+            table_reader_db=table_reader_db,
+            table_reader_table=table_reader_table,
+        )
+        dbconfig = Mock()
+        dbconfig.get_options = Mock(
+            return_value={
+                "mode": "overwrite",
+                "format_": "parquet",
+                "path": "test_folder/historical/entity/feature_set",
+            }
+        )
```

Comment:
Nothing "wrong" here just a tip: you didn't necessarily have to create the table physically. You could just create a temporary view and read it normally with the `table_reader`, passing just the name for the temporary view, the database is not required, so it will be able to read the temporary view as a table without errors.

Personally I try to avoid at maximum to use any context besides the program, even the file system can sometimes get intermittent errors, so the build can break for no reason. Besides, we already have integration tests for the file system, so here I think we could try to simplify in what we can and just test the necessary logic: the running of the pipeline for a given time interval.

Did I make sense to you?

Reply:
Yeap and I've fixed them.